api/queue/work/model_service_deployment_test.go

```diff
@@ -178,6 +190,56 @@ func TestExecuteDeployment(t *testing.T) {
 				return producer
 			}(),
 		},
+		{
+			name:    "Success: with calling webhooks",
+			model:   model,
+			version: version,
+			endpoint: &models.VersionEndpoint{
+				EnvironmentName: env.Name,
+				ResourceRequest: env.DefaultResourceRequest,
+				VersionID:       version.ID,
+				Namespace:       project.Name,
+			},
+			deploymentStorage: func() *mocks.DeploymentStorage {
+				mockStorage := createDefaultMockDeploymentStorage()
+				mockStorage.On("OnDeploymentSuccess", mock.Anything).Return(nil)
+				return mockStorage
+			},
+			storage: func() *mocks.VersionEndpointStorage {
+				mockStorage := &mocks.VersionEndpointStorage{}
+				mockStorage.On("Save", mock.Anything).Return(nil)
+				mockStorage.On("Get", mock.Anything).Return(&models.VersionEndpoint{
+					Environment:     env,
+					EnvironmentName: env.Name,
+					ResourceRequest: env.DefaultResourceRequest,
+					VersionID:       version.ID,
+					Namespace:       project.Name,
+				}, nil)
+				return mockStorage
+			},
+			controller: func() *clusterMock.Controller {
+				ctrl := &clusterMock.Controller{}
+				ctrl.On("Deploy", mock.Anything, mock.Anything).
+					Return(&models.Service{
+						Name:        iSvcName,
+						Namespace:   project.Name,
+						ServiceName: svcName,
+						URL:         url,
+						Metadata:    svcMetadata,
+					}, nil)
+				return ctrl
+			},
+			imageBuilder: func() *imageBuilderMock.ImageBuilder {
+				mockImgBuilder := &imageBuilderMock.ImageBuilder{}
+				return mockImgBuilder
+			},
+			webhookManager: func() webhooks.WebhookManager {
+				webhookManager := webhooks.NewMockWebhookManager(t)
+				webhookManager.On("IsEventConfigured", mock.Anything).Return(true)
+				webhookManager.On("InvokeWebhooks", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
```

Comment:
nit: instead of `mock.Anything` can you use exact parameter that will be passed?

Reply:
Can do for the eventType (and have changed it). But for the third argument/payload (`*versionEndpointRequest`), I don’t think can do it, because the `*jobArgs` [is getting marshaled and unmarshaled](https://github.com/caraml-dev/merlin/blob/75729fd51b63689178882a81ff1a940b01131bab/api/queue/work/model_service_deployment.go#L57-L62) so I can't really compare the pointer address as it will differs from `tt.endpoint`